addons/l10n_ro_edi/models/ciusro_document.py

```diff
@@ -0,0 +1,198 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import io
+import requests
+import zipfile
+
+from lxml import etree
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+NS_UPLOAD = {"ns": "mfp:anaf:dgti:spv:respUploadFisier:v1"}
+NS_STATUS = {"ns": "mfp:anaf:dgti:efactura:stareMesajFactura:v1"}
+NS_HEADER = {"ns": "mfp:anaf:dgti:efactura:mesajEroriFactuta:v1"}
+NS_SIGNATURE = {"ns": "http://www.w3.org/2000/09/xmldsig#"}
+
+
+class CIUSRODocument(models.Model):
+    _name = 'ciusro.document'
+    _description = "Document object for tracking CIUS-RO XML sent to E-Factura"
+    _order = 'datetime DESC, id DESC'
+
+    move_id = fields.Many2one('account.move', required=True)
+    state = fields.Selection(
+        selection=[
+            ('to_send', 'To send'),
+            ('sending', 'Sending'),
+            ('sent', 'Sent'),
+            ('nok', 'Not OK'),
+            ('ok', 'OK'),
+            ('error', 'Error'),
+        ],
+        string='E-Factura Status',
+        default='to_send',
+    )
+    attachment_id = fields.Binary(string='E-Factura Attachment')
+    datetime = fields.Datetime()
+    message = fields.Char()
+    key_loading = fields.Char()
+    key_download = fields.Char()
+    key_signature = fields.Char()
+    key_certificate = fields.Char()
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        documents = super().create(vals_list)
+        for document in documents:
+            document.move_id.l10n_ro_edi_document_ids |= document
+            document.move_id.l10n_ro_edi_active_document_id = document
+        return documents
+
+    def _get_attachment_file_name(self):
+        self.ensure_one()
+        return f"{self.move_id.name.replace('/', '_')}_{self.id}.zip"
+
+    def _handle_error(self, message: str):
+        self.state = 'error'
+        self.message = message
+
+    def _make_request(self, endpoint, xml_data=None, session=None):
+        """ Make an API request to E-Factura and return the response
+        :param endpoint: 'upload' (for sending, requires xml_data) |
+                         'stareMesaj' (for fetching status) |
+                         'descarcare' (for downloading answer)
+        :return: response from E-Factura
+        """
+        self.ensure_one()
+        self.datetime = fields.Datetime.now()
+        self.move_id.l10n_ro_edi_active_document_id = self
+
+        url = f"https://api.anaf.ro/test/FCTEL/rest/{endpoint}"
+        headers = {'Content-Type': 'application/xml',
+                   'Authorization': f'Bearer {self.move_id.company_id.l10n_ro_edi_access_token}'}
```

Comment:
```suggestion
        headers = {
            'Content-Type': 'application/xml',
            'Authorization': f'Bearer {self.move_id.company_id.l10n_ro_edi_access_token}',
        }
```

Reply:
The document should contain the web-service but shouldn't be invoice-specific. What about just passing the company as parameter instead?
The whole method should be `api.model`